### ISSUE TITLE
an actix_web::test macro

### ISSUE THREAD

**Issue by ufoscout** [actix/actix-web#1669]
I am testing actix-web 3.0.0, with this simple file Cargo.toml file:
```toml
[package]
name = "actix_web_3"
version = "0.1.0"
edition = "2018"

[dependencies]
actix-web = "3.0.0"

[dev-dependencies]
actix-rt = "1.1.1"
```

With this cargo file, this test executes as expected:
```rust
#[actix_web::rt::test]
async fn async_test_stub() {
    assert!(true)
}
```

Anyway, if I remove `actix-rt` from the `dev-dependencies`, then the above code fails with this error:
``` 
--> src/lib.rs:2:1
  |
2 | #[actix_web::rt::test]
  | ^^^^^^^^^^^^^^^^^^^^^^ not found in `actix_rt`
  |
  = note: consider importing this struct:
          std::alloc::System
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to previous error
```
This is a bit surprising because I am not directly using `actix-rt`; so, I expected it to work transparently.

**Comment by robjtede** [comment]
At the moment, only the `actix_web::main` macro is supported when used like this and for tests you should still use `actix-rt` directly.

**Comment by ufoscout** [comment]
Hi @robjtede 
is there a technical reason why it does not work the way I am using it? In the release note of actix-web 3 you wrote:

> the entire runtime crate is re-exported as actix_web::rt, too.

So I expected it to work.
I inspected the code and I didn't found any feature flag that could explain the problem. I really don't understand how adding the `actix-rt` dev-dependency could alter the compilation outcome of my code.

**Comment by robjtede** [comment]
Sure, the technical explanation is simple enough:

Where the macros live:
- `actix_web::main` and `actix_web::rt::main` are different macros
- `actix_rt::main` and `actix_web::rt::main` are *the same* macro.

What they expand to:
- `actix_web::main` gets it's macro from `actix-web-codegen`
  - which expands to use `actix_web::rt::System`
- `actix_rt::main` gets it's macro from `actix-macros`
  - which expands to use `actix_rt::System`
  - which won't work if actix_rt isn't in Cargo.toml

So to support the `#[actix_web::test]` use case we'd need to write a "test" macro in actix-web-codegen. We decided against doing so for now because `actix_web` already exports a module called "test".

The change was primarily aimed at getting beginners going quickly; we noticed new folks in the Gitter chat were often asking why there were 2 dependencies needed for the simplest hello world app.

Hope that clears it up :)
